pkg/api/customization/logging/action.go

```diff
@@ -215,14 +226,47 @@ func (h *Handler) dryRunLoggingTarget(apiContext *types.APIContext, level, clust
 	if err != nil {
 		return errors.Wrap(err, "create executor failed")
 	}
-	var stdout, stderr, stdin bytes.Buffer
 
-	dryRunConfig := strings.Replace(string(dryRunConfigBuf), `$`, `\$`, -1)
-	stdin.WriteString(fluentdCommand)
-	stdin.WriteString(fmt.Sprintf(`"%s"`, dryRunConfig))
+	mkDirCmd := bytes.NewBufferString("mkdir -p " + tmpCertDir)
+	if err = remoteExcuteCommand(executor, *mkDirCmd); err != nil {
+		return err
+	}
+
+	var errgrp errgroup.Group
+	if certificate != "" {
+		echoCertificateCmd := bytes.NewBufferString(`echo "` + certificate + `" >> ` + certificatePath)
+		errgrp.Go(func() error { return remoteExcuteCommand(executor, *echoCertificateCmd) })
+	}
+	if clientCert != "" {
+		echoClientCertCmd := bytes.NewBufferString(`echo "` + clientCert + `" >> ` + clientCertPath)
+		errgrp.Go(func() error { return remoteExcuteCommand(executor, *echoClientCertCmd) })
+	}
+	if clientKey != "" {
+		echoClientKeyCmd := bytes.NewBufferString(`echo "` + clientKey + `" >> ` + clientKeyPath)
+		errgrp.Go(func() error { return remoteExcuteCommand(executor, *echoClientKeyCmd) })
+	}
+	if err = errgrp.Wait(); err != nil {
```

Comment:
Would it be better to make less api call? Maybe we can combine all the create command into one call and the test and clean-up into the other one? Or even all those command with in one api call?

Reply:
updated